### ISSUE TITLE
deleted Group Plan task does not look deleted in member's task list

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#9842]
It's not obvious when an assigned Group Plan task has been deleted from the Group. Steps to reproduce:

1. Go to the "Task Board" for a Group Plan you own.
2. Create a task.
3. Assign it to a Group Plan member.
4. In that member's account, sync to make the task appear in the user's Tasks screen.
5. In the Group's Task Board, delete the task.
6. In the member's account, sync again. Observe that the task remains and its appearance is not changed. E.g., the screenshot below is a task in my task list that's been deleted from the Group Plan.

Ideally when a task has been deleted from the Group Plan, we should adjust the appearance of the tasks so that it's clear it's no longer associated with the Group Plan. For example, we could replace "You are assigned to this task     Remove Claim" with something like "Task has been removed from the Group Plan's task list" 

Also, it would be nice if we could have an option to convert the task to a regular user task, as we have for deleted Challenge tasks. A user might want to keep the task and won't want to always see the "has been removed" message.

![screenshot_2018-01-21_16-16-38](https://user-images.githubusercontent.com/1495809/35191453-8b4a7ae2-fec6-11e7-80be-a6fd06c88f4b.png)

FYI @Tressley for design

For the coder who implements this after a design is done, you can identify a deleted Group Plan task by `"broken": "TASK_DELETED"` in the user's copy of it. Example: 
```
{
    "_id": "244ff1aa-d44f-4b1b-91a7-c20b1f16034a",
    "userId": "98089587-1bbe-4de2-92d3-0bbe76513a3f",
    "text": "Alys test",
    "group": {
        "taskId": "9fbec0f2-454e-4c61-a5de-7e12c3614566",
        "id": "f6d914ba-3a24-4cb3-9ab0-ad4bccb1d585",
        "approval": {
            "requested": false,
            "approved": false,
            "required": false
        },
        "assignedUsers": [
            "98089587-1bbe-4de2-92d3-0bbe76513a3f"
        ],
        "broken": "TASK_DELETED"
    },
    .... other stuff ....
}
```

If the user chooses to keep the task the `group` section of the task would need to be changed to the default settings: 
```
    "group": {
        "approval": {
            "requested": false,
            "approved": false,
            "required": false
        },
        "assignedUsers": []
    },
```

**Comment by paglias** [comment]
What if we simply remove the task from the user list? I'm not a big fan of adding another option to keep the task as it adds more complexity to the sync code that we'll have to support and in this case  the task only exists in the user list just because someone from the group plan has assigned it to them so I think it makes sense so remove it

**Comment by Alys** [comment]
There could be good reasons to keep the task.

E.g., a parent gives a child a "clean teeth" task. The child does so well at it that the parent no longer needs to keep monitoring the child. The parent deletes the task from the Group Plan. However the child is so proud of their progress that they want to keep the task with its awesome streak and keep using it to make themselves even better. The child cries when they see the task has been auto-deleted. Why do we make little children cry? :)

Another use case: 
A team leader assigns a task for some important work, say building a server. The team member uses its notes and/or checklist to record steps in the work or important information about the server. They finish the build so the team leader deletes the task from the Group Plan, but the team member still wants to refer to the task to copy out all their useful notes. The team member cries when they see the task has been auto-deleted.

**Comment by shanaqui** [comment]
My assumption was that a deleted group plan task would be treated like a deleted challenge task. As with challenges, someone might want to keep it for their own records, because they want to continue doing it, because they've made awesome progress, etc.

(`quietly removes comment showing self has bad reading comprehension`)

**Comment by paglias** [comment]
So it's okay if we show the task without the group controls (no "you've been assigned") and show the broken megaphone like for broke challenge tasks? @Tressley @Alys

**Comment by Tressley** [comment]
I don't think we should use the broken megaphone icon as the megaphone is respective to Challenges.

**Comment by shanaqui** [comment]
Just thinking out loud... I think that adding a new icon of some sort might be confusing. At least reusing the megaphone would be a clear visual signal to users that this is a task from either a challenge or a group. If we add a new icon, it's one more thing to know -- and the megaphone isn't even universally recognised now, per my experience in Habitica Help.

**Comment by Alys** [comment]
I like the idea of reusing the megaphone. The actions possible for orphaned Group Plan and Challenge tasks are the same so having the same icon makes sense to me. We could improve the current confusion with the icon and reduce future confusion by improving the on-hover text for the megaphone.

**Comment by saraolson** [comment]
the problem with using the megaphone is that it isn't just there when a task is broken. its used to denote a challenge task at all times. its also used within the challenge section on guilds and such when they have no challenges. the megaphone is literally an icon attached to the challenge feature. we could use a similar pattern for group plans, but using the exact icon for something other than challenges is unclear.

**Comment by shanaqui** [comment]
Honestly, from time in Report a Bug and Habitica Help, etc, what's unclear is "lots of different ways to display basically the same thing". The simpler it can be, the better. As a new user, I'd just interpret the megaphone as being "tasks I didn't add myself", not as "challenges specifically" -- if I ever even noticed the challenge icon, which a fair number of people do not (and don't notice the broken megaphone icon either). More stuff to remember or interpret =/= necessarily clearer.

**Comment by veeeeeee** [comment]
Best approach here may be to wait until interactive task board improvements are live, see if that changes user perception of what's going on, and then survey or interview Team plan users in aggregate to ensure we develop a solution that works for them.

**Comment by Tressley** [comment]
This issue looks to have been resolved. If anyone is still experiencing this problem, please feel free to reopen the issue with additional information (OS/browser, steps to reproduce, screen recordings).
